libs/api/domains/identity/src/lib/models/identity.model.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  Field,
+  Extensions,
+  MiddlewareContext,
+  ObjectType,
+  InterfaceType,
+  ID,
+  registerEnumType,
+} from '@nestjs/graphql'
+
+import { NationalRegistryUser } from '@island.is/api/domains/national-registry'
+
+import { IdentityType } from '../identity.type'
+import { Address } from './address.model'
+
+registerEnumType(IdentityType, { name: 'IdentityType' })
+
+@InterfaceType({
+  resolveType(identity: Identity) {
+    switch (identity.type) {
+      case IdentityType.Person:
+        return IdentityPerson
+      case IdentityType.Company:
+        return IdentityCompany
+      default:
+        return null
+    }
+  },
+})
+export class Identity {
+  @Field(() => ID)
+  nationalId!: string
+
+  @Field(() => String)
+  name!: string
+
+  @Field(() => Address, { nullable: true })
+  address?: Address
+
+  @Field(() => IdentityType)
+  type!: IdentityType
+}
+
+const validateIf = ({ context, source }: MiddlewareContext): boolean => {
+  return (context as any).req.user.nationalId !== source.nationalId
+}
+
+@Extensions({
+  validateIf,
+  validFields: [
+    'address',
+    'age',
+    'fullName',
+    'birthday',
+    'type',
+    'name',
+    'nationalId',
```

Comment:
Opt-in is a safer choice for masking out fields, since as new fields come in, they could potentially slip out here.